File: solutions/3418-count-pairs-that-form-a-complete-day-ii/count-pairs-that-form-a-complete-day-ii.go
func countCompleteDayPairs(hours []int) int64 {
    var ans int64 = 0
    count := make([]int, 24)
    
    for _, hour := range hours {
        ans += int64(count[(24 - hour % 24) % 24])
        count[hour % 24]++
    }
    
    return ans
}

File: solutions/3418-count-pairs-that-form-a-complete-day-ii/count-pairs-that-form-a-complete-day-ii.js
/**
 * @param {number[]} hours
 * @return {number}
 */
var countCompleteDayPairs = function(hours) {
    let ans = 0;
    let count = new Array(24).fill(0);
    for (let hour of hours) {
        ans += count[(24 - hour % 24) % 24];
        count[hour % 24]++;
    }
        
    return ans;
};